src/test/java/com/katelocate/menugenerator/RecipeRepositoryTests.java

```diff
@@ -0,0 +1,133 @@
+package com.katelocate.menugenerator;
+
+import com.katelocate.menugenerator.recipe.Recipe;
+import com.katelocate.menugenerator.recipe.RecipeRepository;
+import com.katelocate.menugenerator.recipe.RecipeType;
+
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+
+import static org.hamcrest.Matchers.hasSize;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.server.LocalServerPort;
+
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.PostgreSQLContainer;
+
+import static org.hamcrest.Matchers.equalTo;
+
+import java.util.List;
+
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class RecipeRepositoryTests {
+
+    @LocalServerPort
+    private Integer port;
+
+    static PostgreSQLContainer<?> postgres = new PostgreSQLContainer<>(
+            "postgres:16-alpine"
+    );
+
+    @BeforeAll
+    static void beforeAll() {
+        postgres.start();
+    }
+
+    @AfterAll
+    static void afterAll() {
+        postgres.stop();
+    }
+
+    @DynamicPropertySource
+    static void configureProperties(DynamicPropertyRegistry registry) {
+        registry.add("spring.datasource.url", postgres::getJdbcUrl);
+        registry.add("spring.datasource.username", postgres::getUsername);
+        registry.add("spring.datasource.password", postgres::getPassword);
+    }
+
+    @Autowired
+    RecipeRepository recipeRepository;
+
+    @BeforeEach
+    void setUp() {
+        RestAssured.baseURI = "http://localhost:" + port;
+        recipeRepository.deleteAll();
+    }
+
+    @Test
+    void shouldCreateRecipe() {
+//        ???? Test shouldGetRecipeById covers this functionality
+    }
+
+    @Test
+    void shouldGetAllRecipes() {
+        List<Recipe> recipes = List.of(
+                new Recipe(0, "Breakfast", RecipeType.BREAKFAST, "Scramble eggs"),
+                new Recipe(1, "Dinner", RecipeType.DINNER, "Burrito")
+        );
+        recipeRepository.saveAll(recipes);
+
+        RestAssured.given()
+                .contentType(ContentType.JSON)
+                .when()
+                .get("/api/recipes")
+                .then()
+                .statusCode(200)
+                .body(".", hasSize(2));
+    }
+
+    @Test
+    void shouldGetRecipeById() {
+        recipeRepository.create(new Recipe(0, "Breakfast", RecipeType.BREAKFAST, "Scramble eggs"));
```

Comment:
It feels like `0`, `"Breakfast"` and `"Scramble eggs"` can be moved to constants.

In pythoninc pseudo code I'd do it like that:

```
id = 0
title = "Breakfast"
type = RecipeType.BREAKFAST
body = "Scramble eggs"

recipeRepository.create(new Recipe(id, title, type, body));

RestAssured.given()
    ...
    body(equalTo("{\"id\":" + str(id) + ",\"title\":" + title + ",\"recipeType\":" + str(type) + ",\"body\":" + body + "}"));
```